lib/jsonapi/deserializable/resource.rb

```diff
@@ -3,24 +3,46 @@
 module JSONAPI
   module Deserializable
     class Resource
+      DEFAULT_TYPE_BLOCK = proc { |t| field type: t }
+      DEFAULT_ID_BLOCK   = proc { |i| field id: i }
+      DEFAULT_ATTR_BLOCK = proc { |k, v| field(k.to_sym => v) }
+      DEFAULT_HAS_ONE_REL_BLOCK = proc do |k, _rel, id, type|
+        field "#{k}_type".to_sym => type
+        field "#{k}_id".to_sym   => id
+      end
+      DEFAULT_HAS_MANY_REL_BLOCK = proc do |k, _rel, ids, types|
+        field "#{k}_types".to_sym => types
+        field "#{k}_ids".to_sym   => ids
+      end
+
       include ResourceDSL
 
       class << self
-        attr_accessor :type_block, :id_block, :attr_blocks,
-                      :has_one_rel_blocks, :has_many_rel_blocks
+        attr_accessor :type_block, :id_block, :attr_blocks, :default_attr_block,
+                      :has_one_rel_blocks, :has_many_rel_blocks,
+                      :default_has_one_rel_block,
+                      :default_has_many_rel_block
       end
 
       self.attr_blocks = {}
       self.has_one_rel_blocks  = {}
       self.has_many_rel_blocks = {}
+      self.type_block = DEFAULT_TYPE_BLOCK
+      self.id_block   = DEFAULT_ID_BLOCK
+      self.default_attr_block         = DEFAULT_ATTR_BLOCK
+      self.default_has_one_rel_block  = DEFAULT_HAS_ONE_REL_BLOCK
+      self.default_has_many_rel_block = DEFAULT_HAS_MANY_REL_BLOCK
```

Comment:
Could we make these class methods instead? My thinking is it would be easier to override and stub, and wouldn't need lines 30-34, or 43-45